app/controllers/consolidations/AssociateUcrConfirmationController.scala

```diff
@@ -19,27 +19,25 @@ package controllers.consolidations
 import controllers.actions.{AuthenticatedAction, JourneyRefiner}
 import controllers.storage.FlashKeys
 import javax.inject.Inject
+import models.ReturnToStartException
 import play.api.i18n.I18nSupport
 import play.api.mvc._
-import services.SubmissionService
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import views.html.associate_ucr_confirmation
 
 import scala.concurrent.ExecutionContext
 
 class AssociateUcrConfirmationController @Inject()(
   authenticate: AuthenticatedAction,
+  getJourney: JourneyRefiner,
   mcc: MessagesControllerComponents,
   associateUCRConfirmPage: associate_ucr_confirmation
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport {
 
-  def display: Action[AnyContent] = authenticate { implicit request =>
-    (request2flash.get(FlashKeys.CONSOLIDATION_KIND), request2flash.get(FlashKeys.UCR)) match {
-      case (Some(kind), Some(ucr)) => Ok(associateUCRConfirmPage(kind, ucr))
-      case _                       => Redirect(controllers.routes.ChoiceController.displayPage)
-    }
-
+  def display: Action[AnyContent] = (authenticate andThen getJourney) { implicit request =>
```

Comment:
Interesting approach, If we go with this I think we could create a `getSummary` action which would be used instead of `getJourney` in summary pages.
It could create a 
```
SummaryRequest[T](answers: Map[String, String], request: AuthenticatedRequest[T]) extends WrappedRequest[T](request){
    def get(key: String): String = answers.getOrElse(throw ReturnToStartException)
}
```

Reply:
Thanks.  Only worth pursuing if product team care that the summary data is lost after a screen refresh and/or that the 'choice' is not preserved when starting the next journey

Reply:
I think we can preserve or not preserve choice with either solution